hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/XceiverServerRatis.java

```diff
@@ -552,6 +552,10 @@ private void handlePipelineFailure(RaftGroupId groupId,
       msg = datanode + " is in candidate state for " +
           roleInfoProto.getCandidateInfo().getLastLeaderElapsedTimeMs() + "ms";
       break;
+    case FOLLOWER:
+      msg = datanode + " is in follower state for " +
```

Comment:
Can we update the msg accordingly with the reason. The reason precisely was when installSnapshotFromLeader gets invoked on follower, as leader snapshot doesn't contain any data to replay. All the log entries prior to the snapshot might have been purged. So , follower should not try to install snapshot from leader but can close the pipeline here.

Reply:
@bshashikant Thanks for you review and explanation. I have updated the patch.